core/src/env/api.rs

```diff
@@ -65,11 +65,16 @@ pub fn println(content: &str) {
     ContractEnv::println(content)
 }
 
+/// Deposits an event
+pub fn deposit_event(data: Vec<u8>) {
```

Comment:
For byte buffers at this stage you want to use `&[u8]` instead.
Also the calls in `api.rs` are meant to be a bit more high level if possible.
So I would prefer a solution that is more abstract over events and help users better to get the encoding right (best if no encoding would be needed ofc.)
Might this be improved by: https://github.com/Robbepop/pdsl/issues/42#issuecomment-477219821 ?

Reply:
Would want to talk to you about the second half of your comment in person :)